libs/deer/src/impls/core/array.rs

```diff
@@ -28,9 +28,13 @@ impl<'de, T: Deserialize<'de>, const N: usize> Visitor<'de> for ArrayVisitor<'de
 
         let mut result: Result<(), ArrayAccessError> = Ok(());
 
+        // uninit_assumed_init is fine here, as `[MaybeUninit<T>; N]` as no inhabitants,
+        // the code shown here is also present in 1) the rust docs and 2) as an OK example in the
+        // clippy docs
         #[allow(unsafe_code)]
+        #[allow(clippy::uninit_assumed_init)]
         // SAFETY: this is the same as `MaybeUninit::uninit_array()`, which is still unstable
-        let mut array = unsafe { MaybeUninit::<[MaybeUninit<T>; N]>::uninit().assume_init() };
+        let mut array: [MaybeUninit<T>; N] = unsafe { MaybeUninit::uninit().assume_init() };
```

Comment:
```suggestion
         // SAFETY: `uninit_assumed_init` is fine here, as `[MaybeUninit<T>; N]` as no inhabitants,
         // the code shown here is also present in 1) the rust docs and 2) as an OK example in the
         // clippy docs. The code is the same as in `MaybeUninit::uninit_array()`, which is still unstable
         #[allow(unsafe_code)]
         #[allow(clippy::uninit_assumed_init)]
         let mut array: [MaybeUninit<T>; N] = unsafe { MaybeUninit::uninit().assume_init() };
```

Reply:
fixed in [1e377a1](https://github.com/hashintel/hash/pull/2301/commits/1e377a1068854a018112b80ab10740120bf0e11a)

Reply:
Thanks!